packages/app-desktop/gui/Navigator.jsx

```diff
@@ -4,24 +4,72 @@ const Setting = require('@joplin/lib/models/Setting').default;
 const { connect } = require('react-redux');
 const bridge = require('electron').remote.require('./bridge').default;
 
+function truncStr(str, max = 30) {
+	if (str.length > max) {
+		return `${str.substring(0, max - '..'.length)}..`;
+	}
+	return str;
+}
+
 class NavigatorComponent extends Component {
+	constructor(props) {
+		super(props);
+		this.prevTitle = null;
+	}
 	UNSAFE_componentWillReceiveProps(newProps) {
 		if (newProps.route) {
-			const screenInfo = this.props.screens[newProps.route.routeName];
+			const windowTitle = [];
+			const txt = this.getWindowTitle(newProps);
+			if (txt) { windowTitle.push(txt); }
 			const devMarker = Setting.value('env') === 'dev' ? ' (DEV)' : '';
-			const windowTitle = [`Joplin${devMarker}`];
-			if (screenInfo.title) {
-				windowTitle.push(screenInfo.title());
-			}
+			windowTitle.push(`Joplin${devMarker}`);
 			this.updateWindowTitle(windowTitle.join(' - '));
 		}
 	}
-
+	getWindowTitle(newProps) {
```

Comment:
We need unit test for this function. Perhaps one way would be to move the function to a separate file `getWindowTitle.ts` and have `getWindowTitle.test.ts` next to it. Please see how other test files are setup within the app-desktop package for more information, and if you need any help let me know.

Reply:
I'm new to unit testing, but I'm learning more about it. If I need any help I will sure reach out.